programs/vote/src/vote_state/mod.rs

```diff
@@ -382,13 +407,196 @@ impl VoteState {
         }
     }
 
+    fn check_update_vote_state_slots_are_valid(
+        &self,
+        vote_state_update: &mut VoteStateUpdate,
+        slot_hashes: &[(Slot, Hash)],
+    ) -> Result<(), VoteError> {
+        // If the vote state update is not new enough, return
+        if let Some(last_vote_slot) = self.votes.back().map(|lockout| lockout.slot) {
+            if vote_state_update.lockouts.back().unwrap().slot <= last_vote_slot {
+                return Err(VoteError::VoteTooOld);
+            }
+        }
+
+        if vote_state_update.lockouts.is_empty() {
```

Comment:
should put this before the previous check so that `vote_state_update.lockouts.back().unwrap()` can't fail.

Reply:
ah good catch, thanks